File: src/main/java/com/example/demo/dtos/UserRegisterDto.java
package com.example.demo.dtos;

import com.example.demo.model.User;

/**
 * Created by Emirhan Doğandemir at 29.09.2021
 */
public class UserRegisterDto {
    private String username;
    private String password;
    private String email;

    public User getUserFromDto() {
        User user = new User();
        user.setUsername(username);
        user.setPassword(password);
        user.setEmail(email);
        return user;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: src/main/java/com/example/demo/dtos/UserRegisterDto.kt
package com.example.demo.dtos

import com.example.demo.model.Credit
import com.example.demo.model.CreditNote
import com.example.demo.model.Role
import java.math.BigDecimal
import java.util.*
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

/**
 * Created by İbrahim Başar Yargıcı and Emirhan Doğandemir at 2.10.2021
 */
data class UserRegisterDto(
    val id: Long? = 0,

    val firstname: String = "Basar",

    val lastname: String = "Emirhan",

    @field:NotNull
    @field:Email
    var email: String = "be@gmail.com",

    @field:NotBlank
    @field:NotNull
    @field:Size(max = 20)
    var username: String,

    @field:NotBlank
    @field:NotNull
    @field:Size(
        min = 6,
        max = 20,
        message = "Password must be between 6 and 20 characters"
    )
    var password: String,

    val balance: BigDecimal = BigDecimal.valueOf(900),

    val creditNote: CreditNote? = CreditNote.C,

    var credit: Credit?,

    var roles: Set<Role>? = HashSet()
)